src/include/log_inline.h

```diff
@@ -19,8 +19,8 @@ __wt_log_cmp(WT_LSN *lsn1, WT_LSN *lsn2)
      * Read LSNs into local variables so that we only read each field once and all comparisons are
      * on the same values.
      */
-    l1 = ((volatile WT_LSN *)lsn1)->file_offset;
```

Comment:
The semantics have changed a bit here as `READ_ONCE` performs an interim cast into `volatile uint64_t` instead of `volatile WT_LSN *`, but the end result looks the same to me.

Reply:
I'd say yes since the end result is `l1` and `l2` get cached locally which prevents any invented reads in the following line `(l1 < l2 ? -1 : (l1 > l2 ? 1 : 0))`. This aligns with the comment above.

The risk we face is if the volatile casts are leaking outside the scope of this function and preventing fuses in the callers, but I don't see anything indicating that from [their original introduction](https://github.com/wiredtiger/wiredtiger/pull/2083).